File: app/src/main/java/com/example/android/architecture/services/ServiceProvider.java
package com.example.android.architecture.services;

import com.example.android.architecture.data.remote.apis.ApiProvider;
import com.example.android.architecture.data.remote.apis.PostApi;

/**
 * Created by YaoHaitao on 2018/3/12.
 */

public class ServiceProvider {

    public static PostService makePostService() {
        return new PostService(ApiProvider.getInstance().makeApi(PostApi.class));
    }

}

File: app/src/main/java/com/example/android/architecture/utils/ServiceProvider.kt
package com.example.android.architecture.utils

import com.example.android.architecture.services.PostService
import com.example.android.architecture.services.UserService

object ServiceProvider {

  fun makePostService() = PostService(ApiProvider.makePostApi())

  fun makeUserService() = UserService(ApiProvider.makeUserApi())
}




